strawberryfields/ops.py

```diff
@@ -1910,6 +1910,23 @@ def _decompose(self, reg, **kwargs):
         ]
 
 
+class sMZgate(Gate):
+    r"""Symmetric Mach-Zehnder interferometer"""
+    ns = 2
+
+    def __init__(self, phi_in, phi_ex):
+        super().__init__([phi_in, phi_ex])
+
+    def _decompose(self, reg, **kwargs):
+        # into local phase shifts and two 50-50 beamsplitters
+        return [
+            Command(BSgate(np.pi / 4, np.pi / 2), reg),
+            Command(Rgate(self.p[1] - np.pi / 2), reg[1]),
+            Command(Rgate(self.p[0] - np.pi / 2), reg[0]),
+            Command(BSgate(np.pi / 4, np.pi / 2), reg),
+        ]
+
+
```

Comment:
this is also very nice. Great decomposition work!

Reply:
thanks :D

Reply:
...this took us longer than i'd like to admit!